test_cases/stdlib/builtins/test_pow.py

```diff
@@ -0,0 +1,44 @@
+from decimal import Decimal
+from fractions import Fraction
+from typing import Any, NoReturn
+from typing_extensions import Literal, assert_type
+
+assert_type(pow(1, 0), Literal[1])  # See #7163
+assert_type(pow(1, 0, None), Literal[1])  # See #7163
+assert_type(pow(2, 4, 0), NoReturn)
+assert_type(pow(2, 4), int)
+assert_type(pow(5, -7), float)
+assert_type(pow(2, 4, 5), int)  # pow(<smallint>, <smallint>, <smallint>)
+assert_type(pow(2, 35, 3), int)  # pow(<smallint>, <bigint>, <smallint>)
+assert_type(pow(4.6, 8), float)
+assert_type(pow(5.1, 4, None), float)
+assert_type(pow(complex(6), 6.2), complex)
+assert_type(pow(complex(9), 7.3, None), complex)
+assert_type(pow(Fraction(), 4, None), Fraction)
+assert_type(pow(Fraction(3, 7), complex(1, 8)), complex)
+assert_type(pow(complex(4, -8), Fraction(2, 3)), complex)
+assert_type(pow(Decimal("1.0"), Decimal("1.6")), Decimal)
+assert_type(pow(Decimal("1.0"), Decimal("1.0"), Decimal("1.0")), Decimal)
+assert_type(pow(Decimal("4.6"), 7, None), Decimal)
+assert_type((4).__pow__(7, 4), int)
```

Comment:
Why are you calling `__pow__` directly here? That doesn't seem idiomatic.

Reply:
Honest answer: no idea. (I've had these test cases sitting around in my back pocket for [a long time](https://github.com/python/typeshed/pull/6325#issue-1055588877).) I'll change it :)

Reply:
Hmm, actually, how do you test the `__pow__` overloads on `int` with three arguments with out accessing `__pow__` directly? If you do `pow(x, y, z)`, you can't be sure you're not testing the overloads for the builtin function `pow()` (rather than the overloads on `int`), and if you do `x ** y`, you can't test the three-argument form.

Reply:
On third thought, it doesn't matter at all which method/function we're testing if the end result is correct. So you're right, these tests are weird; I'll change them.